bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioClient.java

```diff
@@ -377,6 +380,77 @@ public void setVolume(AbstractAudioDeviceConfig item, int vol) {
         item.setVolume(Math.round(100f / 65536f * vol));
     }
 
+    /**
+     * Locate or load (if needed) the simple protocol tcp module for the given sink
+     * and returns the port.
+     * The module loading (if needed) will be tried several times, on a new random port each time.
+     *
+     * @param item the sink we are searching for
+     * @param simpleTcpPortPref the port to use if we have to load the module
+     * @return the port on which the module is listening
+     */
+    public Optional<Integer> loadModuleSimpleProtocolTcpIfNeeded(AbstractAudioDeviceConfig item,
+            Integer simpleTcpPortPref) {
+        int currentTry = 0;
+        int simpleTcpPortToTry = simpleTcpPortPref;
+        do {
+            Optional<Integer> simplePort = findSimpleProtocolTcpModule(item);
+
+            if (simplePort.isPresent()) {
+                return simplePort;
+            } else {
+                sendRawCommand("load-module module-simple-protocol-tcp sink=" + item.getPaName() + " port="
+                        + simpleTcpPortToTry);
+                simpleTcpPortToTry = new Random().nextInt(64512) + 1024; // a random port above 1024
+            }
+            try { // let time for the module to load
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+            }
```

Comment:
I just noticed that `loadModuleSimpleProtocolTcpIfNeeded()` is called from within `initialize()` in the Thing handler. `initialize()` should return fast. You could enclose the entire `if` in `initialize()` with `scheduler.submit()` to execute it asynchronously.

Also, the `InterruptedException` should be cought outside the loop or even thrown by this method to handle it as high as possible. 
The `InterruptedException` is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done.

Reply:
It's all good now, thank you.